app/scripts/west_commands/metadata.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2021 The ZMK Contributors
+# SPDX-License-Identifier: MIT
+'''Metadata command for ZMK.'''
+
+import glob
+import json
+from jsonschema import validate, ValidationError
+import os
+import yaml
+from textwrap import dedent            # just for nicer code indentation
+
+from west.commands import WestCommand
+from west import log                   # use this for user output
+
+
+class Metadata(WestCommand):
+    def __init__(self):
+        super().__init__(
+            'metadata',  # gets stored as self.name
+            'ZMK hardware metadata commands',  # self.help
+            # self.description:
+            dedent('''Operate on the board/shield metadata.'''))
+
+    def do_add_parser(self, parser_adder):
+        parser = parser_adder.add_parser(self.name,
+                                         help=self.help,
+                                         description=self.description)
+
+        parser.add_argument('subcommand', default="check",
+                            help='The subcommand to run. Defaults to "check".', nargs="?")
+        return parser           # gets stored as self.parser
+
+    def do_run(self, args, unknown_args):
+        schema = json.load(open("../schema/hardware-metadata.schema.json", 'r'))
+        failure = False
+        for file in glob.glob("boards/**/*.zmk.yml", recursive=True):
+            print("Validating: " + file)
+            with open(file, 'r') as stream:
+                try:
+                    metadata = yaml.safe_load(stream)
+                    ret = validate(metadata, schema)
+                except yaml.YAMLError as exc:
+                    failure = True
```

Comment:
IMO it would be simpler to just
```python
print(exc)
sys.exit(1)
```
instead of having a common exit point. You've already hit an error, so you should probably bail immediately.

Alternatively, you could wrap all of this up in another function and have it re-raise the exceptions with added info, then have a single handler that prints the error and exits. It might look something like this:

```python
class MetadataError(Exception):
    pass

def do_run(self, args, unknown_args):
    try:
        self.validate_files()
    except MetadataError as e:
        print(e)
        sys.exit(1)
        
def validate_files(self):
    # ...
    try:
        # ...
    except yaml.YAMLError as e:
        # Use .format() instead of f-strings if we care about Python < 3.6
        raise MetadataError(f'Failed loading YAML file "{file}": {e}') from e
    except ValidationError as e:
        raise MetadataError(f'Failed validation of file "{file}": {e}') from e
```
...which I kinda like because all the logic for printing an error is in one spot instead of spread across multiple exception handlers.

Reply:
Oh, the point is to continue, so we can check *all* the files. Then if any fail... we return a non-zero exit code.

Reply:
We specifically *don't* want to exit early.